sky/job/controller.py

```diff
@@ -249,7 +249,8 @@ def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
                 cluster_status_str = ('' if cluster_status is None else
                                       f' (status: {cluster_status.value})')
                 logger.info(
-                    f'Cluster is preempted{cluster_status_str}. Recovering...')
+                    f'Cluster is preempted or failed{cluster_status_str}. '
+                    'Recovering...')
```

Comment:
Update all comments in this func w.r.t. "preempted"? To account for the new cases.

Also, shall we test an on-demand managed job's cluster failing + recovering?

Reply:
I tested the manual termination and it works correctly as we did for spot preemptions.

For the simulation of a maintenance event, it seems a CPU instance will not experience any downtime as GCP will migrate the whole instance, which works correctly with our managed job.

For the GPU instance, it seems the `gcloud compute instances simulate-maintenance-event` will hang forever, though no obvious reason could be found online.